corehq/apps/userreports/models.py

```diff
@@ -660,26 +667,22 @@ def all(cls, use_server_filter=True):
 
     @classmethod
     def by_domain(cls, domain):
-        return [ds for ds in cls.all() if ds.domain == domain]
+        return [
+            cls._get_datasource_config(wrapped, dom)
```

Comment:
Guessing htis function isn't actually used much? Otherwise we could possibly store it keyed by domain and id if we could pass in domain to by_id